### ISSUE TITLE
Option for truncated BPTT

### ISSUE THREAD

**Issue by mjwillson** [bartvm/blocks#592]
Wondering if this might be possible in bricks.recurrent. It looks like `theano.scan` supports a `truncate_gradient=N` option, can you see any potential problems with just allowing that to be passed on to theano from a recurrent brick? If not I'll see if I can add it.
Cheers!

**Comment by EderSantana** [comment]
I don't think it is already implemented, otherwise it should appear [here](https://github.com/bartvm/blocks/blob/master/blocks/bricks/recurrent.py#L204-L209). To add that, you can try to "pop" `truncate_gradient` from `kwargs`. I think I could use that feature too.

**Comment by rizar** [comment]
We definitely should support it. One way is to make it a keyword argument of `recurrent_apply`, but I don't like it too much because this a bit lower than Blocks level of abstraction. Another way is simply add a `scan_kwargs` keyword argument for the recurrent calls, using which one could pass whatever he likes to the actual `theano.scan` call. I would prefer the latter.

**Comment by mjwillson** [comment]
Yeah, it feels like it's not really part of the model structure / compute graph but an optimisation setting. But theano wants it passed to scan. I guess for now I'm not too picky how it's done, just want to see if I can use it to speed up training a bit. I'll try out what you suggest, cheers

**Comment by EderSantana** [comment]
I liked @rizar's suggestion better as well.
